content/articles/implementation-of-fuzzy-logic-in-matlab/index.md

```diff
@@ -0,0 +1,99 @@
+
+![fuzzy concept](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_one.png)
+
+As shown in the diagram above, you convert the crisp input with precise value with a particular rating to fuzzy variables with fuzzy logic. This process is known as fuzzification. Then, you apply the fuzzy logic rules to the fuzzy variables to create a new fuzzy number. Finally, you convert the fuzzy number back to a crisp, specific output in a process called fuzzification.
+
+### Fuzzy logic approach
+Suppose we poll 100 bankers and ask them to assign the transition from bad to neutral credit and change from neutral to good. More than likely, in this case, there are going to be a different value for this. Let us assume that every banker thought 750 and above is good credit, but only half thought 700 was good, and none thought 650 was good. It traces out the particular membership function for good credit. We can do the same for neutral credit and bad credit. Here, we will end up with three overlapping membership functions.
+
+![fuzzy logic](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_two.png)
+
+Now, we can see that the statement's truth does not belong to a single set but can belong to multiple sets based on the degree of the truth.
+
+### Implementation of the fuzzy logic in Matlab
+Consider a very simple example;
+We need to control the speed of a motor by changing the input voltage when a set point is defined. If the motor runs fast for some reason, we need to slow it down by reducing the input voltage. Conversely, if the motor slows below the set point, the input voltage must be increased so that the motor reaches the set point.
+let the input status word be;
+Too slow
+Just right
+Too fast
+
+let the output action words be;
+Less voltage(slow down)
+No change
+More voltage(speed up)
+
+The rules base are;
+1. if the motor is running too slow, then more voltage.
+2. If the motor speed is right, then no change.
+3. If the motor speed is too fast, then the voltage is less.
+
+We will define all these in a fuzzy system and create a file for this. The first step is to open the fuzzy logic toolbox. To do this, execute the `fuzzy` command in the command window. When you open the logic toolbox, a new window shown below will pop up.
+
+![fuzzy toolbox](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_three.jpeg)
+
+Here, we have three blocks, is, input, Mamdani and output. In the input, this is where we input our data and for the fuzzification process. The output section, which displays the output results, acts as a defuzzification. Finally, the Mamdani is for the rule-based block. What it means is that you define your rules in this box. Now double-click on the input to open it up. When you do so, a window shown below opens up.
+
+![input window](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_four.jpeg)
+*figure1.0*
```

Comment:
If possible, add these for other images as well.

(Good image description would be needed)